static/js/metrics-explorer.js

```diff
@@ -52,17 +52,23 @@ $(document).ready(function() {
 });
 
 
-function metricsExplorerDatePickerHandler(evt) {
+async function metricsExplorerDatePickerHandler(evt) {
     evt.preventDefault();
     $.each($(".range-item.active"), function () {
         $(this).removeClass('active');
     });
+    var selectedId = $(evt.currentTarget).attr("id");
     $(evt.currentTarget).addClass('active');
-    datePickerHandler($(this).attr('id'), "now", $(this).attr('id'))
+    datePickerHandler(selectedId, "now", selectedId);
+    const queryElement = $('#metrics-queries')
+    const metricNames = await getMetricNames();
+    metricNames.metricNames.sort();
+    queryElement.find('.metrics').val(metricNames.metricNames[0]); 
+    getFunctions();
```

Comment:
We don't need to call the getFunctions every time we change the time range since it will always be the same.